src/tasks/helpers/finders/GlobalCoreConfigurationFinder.ts

```diff
@@ -0,0 +1,57 @@
+import {Syntax} from "esprima";
+import * as ESTree from "estree";
+
+import {EMPTY_FINDER_RESULT, Finder, FinderResult} from "../../../dependencies";
+import {SapUiDefineCall} from "../../../util/SapUiDefineCall";
+
+const getPropertyValue = (node: ESTree.Node) => {
+	if (node.type === Syntax.Identifier) {
+		return node.name;
+	}
+	return "";
+};
+
+class FunctionExtensionFinder implements Finder {
+	/**
+	 * Finds expression that matches one of the following conditions
+	 * <ul>
+	 * <li>sap.ui.getCore().getConfiguration()</li>
+	 * </ul>
+	 */
+	find(
+		node: ESTree.Node,
+		config: {},
+		sConfigName: string,
+		defineCall: SapUiDefineCall
+	): FinderResult {
+		const oInterestingNode = node;
+		const bGetCoreGetConfigurationCall =
+			oInterestingNode.type === Syntax.CallExpression &&
+			oInterestingNode.callee.type === Syntax.MemberExpression &&
+			oInterestingNode.arguments.length === 0 &&
+			getPropertyValue(oInterestingNode.callee.property) ===
+				"getConfiguration" &&
```

Comment:
Those 4 lines are the same as the next 4 lines. They can be put into a separate function that accepts two parameters:

- node
- function name

And the new function can be called:

- isNamedFunctionCall(oInterestingNode, "getConfiguration")
- isNamedFunctionCall(oInterestingNode.callee.object, "getCore")

Reply:
good point